subsys/app_memory/app_memdomain.c

```diff
@@ -92,6 +92,21 @@ void app_calc_size(void)
  */
 void appmem_init_app_memory(void)
 {
+#ifdef CONFIG_ARC
+	/*
+	 * appmem_init_app_memory will access all partitions
+	 * For CONFIG_ARC_MPU_VER == 3, these partitions are not added
+	 * into MPU now, so need to disable mpu first to do app_bss_zero()
+	 */
+	extern void arc_core_mpu_disable(void);
```

Comment:
Ugh.
Is there any way we can do this without hacking in ARC-specific calls into what is supposed to be platform-agnostic subsystem code?